app/models/poll/elector.rb

```diff
@@ -0,0 +1,95 @@
+class Poll
+  class Elector < ApplicationRecord
+    before_validation :sanitize
+    before_validation :set_user
+
+    belongs_to :electoral_college,
+      class_name: "Poll::ElectoralCollege",
+      foreign_key: "poll_electoral_college_id"
+
+    belongs_to :user
+
+    validates :document_type, presence: true
+    validates :document_type,
+      inclusion: { in: User::DOCUMENT_TYPES, allow_blank: true }
+
+    validates :document_number, presence: true
+    validates :document_number,
+      uniqueness: { scope: [:electoral_college, :category] }
+
+    validate :valid_document_number
+
+    scope :user_not_found, -> { where(user_found: false) }
+    scope :by_category, ->(category) { where(category: category) }
+    scope :active_electoral_college, -> do
+      joins(:electoral_college).where(poll_electoral_colleges: { active: true })
+    end
+    scope :by_document, ->(document_type, document_number) do
+      where(document_type: document_type, document_number: document_number)
+    end
+    scope :by_electoral_college, ->(electoral_college) do
+      where(electoral_college: electoral_college)
+    end
+
+    def self.search(terms)
+      Elector.where("document_number ILIKE ?", "%#{terms}%")
+    end
+
+    def self.quick_search(terms)
+      if terms.blank?
+        Elector.none
+      else
+        search(terms)
+      end
+    end
+
+    private
+
+    def valid_document_number
+      return if document_number.blank? || document_type.blank?
+
+      if invalid_cpf? || invalid_rnm?
+        message = I18n.t(
+          :invalid_number,
+          scope: 'activerecord.errors.models.user.attributes.cpf'
+        )
+        errors.add(:document_number, message)
+      end
+    end
+
+    def sanitize
+      self.document_type = document_type&.downcase&.strip
+      self.document_number = clean_document_number
+      self.category = category&.capitalize&.strip
+    end
+
+    def clean_document_number
+      return unless document_number
+      self.document_number = document_number.gsub(/[^a-z0-9]+/i, "").upcase
+    end
+
+    def set_user
+      if user = find_user
+        self.user = user
+        self.user_found= true
```

Comment:
Incluir espaço na atribuição
Também é necessário o atributo user_found ?
Se for chamado só no scope pode ser substituído por self.user.present?
Se não, parece melhor criar um método user_found? retornando true ou false